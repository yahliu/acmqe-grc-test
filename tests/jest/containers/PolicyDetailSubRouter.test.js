/* Copyright (c) 2020 Red Hat, Inc. */
'use strict'

import React from 'react'
import PolicyDetailSubRouter from '../../../src-web/containers/PolicyDetailSubRouter'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import { MemoryRouter, Route } from 'react-router-dom'
import * as reducers from '../../../src-web/reducers'
import { createStore, combineReducers, applyMiddleware, compose } from 'redux'
import thunkMiddleware from 'redux-thunk'
import GrcApolloClient from '../../../lib/client/apollo-client'
import { ApolloProvider } from 'react-apollo'
import { Provider } from 'react-redux'

const preloadedState = window.__PRELOADED_STATE__
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
const middleware = [thunkMiddleware]
const store = createStore(combineReducers(reducers), preloadedState, composeEnhancers(
  applyMiddleware(...middleware)
))
const location = {
  'pathname': '/multicloud/policies/all/test-namespace/test-policy',
  'search': '',
  'hash': '',
  'key': 'ngh5of',
  'state': undefined
}
const match = {
  'path': '/multicloud/policies/all/:namespace/:name/:tab?'
}
const resourceType = {
  'name': 'HCMCompliance',
  'query': 'ALL_POLICIES'
}
const tabs = [
  'detail',
  'status',
  'yaml'
]
const updateSecondaryHeader = jest.fn()

describe('PolicyDetailsSubRouter container test', () => {
  it('renders as expected', () => {
    const component = mount(
      <ApolloProvider client={GrcApolloClient.getGrcClient()}>
        <Provider store={store}>
          <MemoryRouter initialEntries={[location]}>
            <Route
              path={match.path}
              render={() =>
              <PolicyDetailSubRouter
                resourceType={resourceType}
                tabs={tabs}
                updateSecondaryHeader={updateSecondaryHeader}
              />}
            />
          </MemoryRouter>
        </Provider>
      </ApolloProvider>
    )
    expect(toJson(component.find(PolicyDetailSubRouter))).toMatchSnapshot()
  })
})
