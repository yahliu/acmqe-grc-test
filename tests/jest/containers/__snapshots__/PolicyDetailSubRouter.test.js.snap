// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicyDetailsSubRouter container test renders as expected 1`] = `
<withRouter(Connect(PolicyDetailSubRouter))
  resourceType={
    Object {
      "name": "HCMCompliance",
      "query": "ALL_POLICIES",
    }
  }
  tabs={
    Array [
      "detail",
      "status",
      "yaml",
    ]
  }
  updateSecondaryHeader={[MockFunction]}
>
  <Connect(PolicyDetailSubRouter)
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "ngh5of",
            "pathname": "/multicloud/policies/all/test-namespace/test-policy",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "ngh5of",
          "pathname": "/multicloud/policies/all/test-namespace/test-policy",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    location={
      Object {
        "hash": "",
        "key": "ngh5of",
        "pathname": "/multicloud/policies/all/test-namespace/test-policy",
        "search": "",
        "state": undefined,
      }
    }
    match={
      Object {
        "isExact": true,
        "params": Object {
          "name": "test-policy",
          "namespace": "test-namespace",
          "tab": undefined,
        },
        "path": "/multicloud/policies/all/:namespace/:name/:tab?",
        "url": "/multicloud/policies/all/test-namespace/test-policy",
      }
    }
    resourceType={
      Object {
        "name": "HCMCompliance",
        "query": "ALL_POLICIES",
      }
    }
    tabs={
      Array [
        "detail",
        "status",
        "yaml",
      ]
    }
    updateSecondaryHeader={[MockFunction]}
  >
    <PolicyDetailSubRouter
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "ngh5of",
              "pathname": "/multicloud/policies/all/test-namespace/test-policy",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "ngh5of",
            "pathname": "/multicloud/policies/all/test-namespace/test-policy",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      location={
        Object {
          "hash": "",
          "key": "ngh5of",
          "pathname": "/multicloud/policies/all/test-namespace/test-policy",
          "search": "",
          "state": undefined,
        }
      }
      match={
        Object {
          "isExact": true,
          "params": Object {
            "name": "test-policy",
            "namespace": "test-namespace",
            "tab": undefined,
          },
          "path": "/multicloud/policies/all/:namespace/:name/:tab?",
          "url": "/multicloud/policies/all/test-namespace/test-policy",
        }
      }
      resourceType={
        Object {
          "name": "HCMCompliance",
          "query": "ALL_POLICIES",
        }
      }
      tabs={
        Array [
          "detail",
          "status",
          "yaml",
        ]
      }
      updateSecondaryHeader={[Function]}
    >
      <Page>
        <div
          className="page-content-container"
          role="main"
        >
          <withRouter(Connect(PolicyDetailsTab))
            policyName="test-policy"
            policyNamespace="test-namespace"
            resourceType={
              Object {
                "name": "HCMCompliance",
                "query": "ALL_POLICIES",
              }
            }
            tabs={
              Array [
                "detail",
                "status",
                "yaml",
              ]
            }
            url="/multicloud/policies/all/test-namespace/test-policy"
          >
            <Connect(PolicyDetailsTab)
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "key": "ngh5of",
                      "pathname": "/multicloud/policies/all/test-namespace/test-policy",
                      "search": "",
                      "state": undefined,
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "ngh5of",
                    "pathname": "/multicloud/policies/all/test-namespace/test-policy",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "key": "ngh5of",
                  "pathname": "/multicloud/policies/all/test-namespace/test-policy",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {
                    "name": "test-policy",
                    "namespace": "test-namespace",
                    "tab": undefined,
                  },
                  "path": "/multicloud/policies/all/:namespace/:name/:tab?",
                  "url": "/multicloud/policies/all/test-namespace/test-policy",
                }
              }
              policyName="test-policy"
              policyNamespace="test-namespace"
              resourceType={
                Object {
                  "name": "HCMCompliance",
                  "query": "ALL_POLICIES",
                }
              }
              tabs={
                Array [
                  "detail",
                  "status",
                  "yaml",
                ]
              }
              url="/multicloud/policies/all/test-namespace/test-policy"
            >
              <PolicyDetailsTab
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "canGo": [Function],
                    "createHref": [Function],
                    "entries": Array [
                      Object {
                        "hash": "",
                        "key": "ngh5of",
                        "pathname": "/multicloud/policies/all/test-namespace/test-policy",
                        "search": "",
                        "state": undefined,
                      },
                    ],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "index": 0,
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "key": "ngh5of",
                      "pathname": "/multicloud/policies/all/test-namespace/test-policy",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "key": "ngh5of",
                    "pathname": "/multicloud/policies/all/test-namespace/test-policy",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {
                      "name": "test-policy",
                      "namespace": "test-namespace",
                      "tab": undefined,
                    },
                    "path": "/multicloud/policies/all/:namespace/:name/:tab?",
                    "url": "/multicloud/policies/all/test-namespace/test-policy",
                  }
                }
                policyName="test-policy"
                policyNamespace="test-namespace"
                resourceType={
                  Object {
                    "name": "HCMCompliance",
                    "query": "ALL_POLICIES",
                  }
                }
                tabs={
                  Array [
                    "detail",
                    "status",
                    "yaml",
                  ]
                }
                updateSecondaryHeader={[Function]}
                url="/multicloud/policies/all/test-namespace/test-policy"
              >
                <Query
                  notifyOnNetworkStatusChange={true}
                  pollInterval={10000}
                  query={
                    Object {
                      "definitions": Array [
                        Object {
                          "directives": Array [],
                          "kind": "OperationDefinition",
                          "name": Object {
                            "kind": "Name",
                            "value": "getSinglePolicy",
                          },
                          "operation": "query",
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": Object {
                                  "kind": "Name",
                                  "value": "items",
                                },
                                "arguments": Array [
                                  Object {
                                    "kind": "Argument",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "name",
                                    },
                                    "value": Object {
                                      "kind": "Variable",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "name",
                                      },
                                    },
                                  },
                                  Object {
                                    "kind": "Argument",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "namespace",
                                    },
                                    "value": Object {
                                      "kind": "Variable",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "namespace",
                                      },
                                    },
                                  },
                                ],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "compliances",
                                },
                                "selectionSet": Object {
                                  "kind": "SelectionSet",
                                  "selections": Array [
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "raw",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "metadata",
                                      },
                                      "selectionSet": Object {
                                        "kind": "SelectionSet",
                                        "selections": Array [
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "creationTimestamp",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "name",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "namespace",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "annotations",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "selfLink",
                                            },
                                            "selectionSet": undefined,
                                          },
                                        ],
                                      },
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "placementPolicies",
                                      },
                                      "selectionSet": Object {
                                        "kind": "SelectionSet",
                                        "selections": Array [
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "metadata",
                                            },
                                            "selectionSet": Object {
                                              "kind": "SelectionSet",
                                              "selections": Array [
                                                Object {
                                                  "alias": undefined,
                                                  "arguments": Array [],
                                                  "directives": Array [],
                                                  "kind": "Field",
                                                  "name": Object {
                                                    "kind": "Name",
                                                    "value": "annotations",
                                                  },
                                                  "selectionSet": undefined,
                                                },
                                                Object {
                                                  "alias": undefined,
                                                  "arguments": Array [],
                                                  "directives": Array [],
                                                  "kind": "Field",
                                                  "name": Object {
                                                    "kind": "Name",
                                                    "value": "name",
                                                  },
                                                  "selectionSet": undefined,
                                                },
                                                Object {
                                                  "alias": undefined,
                                                  "arguments": Array [],
                                                  "directives": Array [],
                                                  "kind": "Field",
                                                  "name": Object {
                                                    "kind": "Name",
                                                    "value": "namespace",
                                                  },
                                                  "selectionSet": undefined,
                                                },
                                                Object {
                                                  "alias": undefined,
                                                  "arguments": Array [],
                                                  "directives": Array [],
                                                  "kind": "Field",
                                                  "name": Object {
                                                    "kind": "Name",
                                                    "value": "creationTimestamp",
                                                  },
                                                  "selectionSet": undefined,
                                                },
                                                Object {
                                                  "alias": undefined,
                                                  "arguments": Array [],
                                                  "directives": Array [],
                                                  "kind": "Field",
                                                  "name": Object {
                                                    "kind": "Name",
                                                    "value": "selfLink",
                                                  },
                                                  "selectionSet": undefined,
                                                },
                                              ],
                                            },
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "clusterLabels",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "status",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "raw",
                                            },
                                            "selectionSet": undefined,
                                          },
                                        ],
                                      },
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "placementBindings",
                                      },
                                      "selectionSet": Object {
                                        "kind": "SelectionSet",
                                        "selections": Array [
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "metadata",
                                            },
                                            "selectionSet": Object {
                                              "kind": "SelectionSet",
                                              "selections": Array [
                                                Object {
                                                  "alias": undefined,
                                                  "arguments": Array [],
                                                  "directives": Array [],
                                                  "kind": "Field",
                                                  "name": Object {
                                                    "kind": "Name",
                                                    "value": "name",
                                                  },
                                                  "selectionSet": undefined,
                                                },
                                                Object {
                                                  "alias": undefined,
                                                  "arguments": Array [],
                                                  "directives": Array [],
                                                  "kind": "Field",
                                                  "name": Object {
                                                    "kind": "Name",
                                                    "value": "namespace",
                                                  },
                                                  "selectionSet": undefined,
                                                },
                                                Object {
                                                  "alias": undefined,
                                                  "arguments": Array [],
                                                  "directives": Array [],
                                                  "kind": "Field",
                                                  "name": Object {
                                                    "kind": "Name",
                                                    "value": "creationTimestamp",
                                                  },
                                                  "selectionSet": undefined,
                                                },
                                                Object {
                                                  "alias": undefined,
                                                  "arguments": Array [],
                                                  "directives": Array [],
                                                  "kind": "Field",
                                                  "name": Object {
                                                    "kind": "Name",
                                                    "value": "selfLink",
                                                  },
                                                  "selectionSet": undefined,
                                                },
                                              ],
                                            },
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "placementRef",
                                            },
                                            "selectionSet": Object {
                                              "kind": "SelectionSet",
                                              "selections": Array [
                                                Object {
                                                  "alias": undefined,
                                                  "arguments": Array [],
                                                  "directives": Array [],
                                                  "kind": "Field",
                                                  "name": Object {
                                                    "kind": "Name",
                                                    "value": "name",
                                                  },
                                                  "selectionSet": undefined,
                                                },
                                                Object {
                                                  "alias": undefined,
                                                  "arguments": Array [],
                                                  "directives": Array [],
                                                  "kind": "Field",
                                                  "name": Object {
                                                    "kind": "Name",
                                                    "value": "kind",
                                                  },
                                                  "selectionSet": undefined,
                                                },
                                              ],
                                            },
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "subjects",
                                            },
                                            "selectionSet": Object {
                                              "kind": "SelectionSet",
                                              "selections": Array [
                                                Object {
                                                  "alias": undefined,
                                                  "arguments": Array [],
                                                  "directives": Array [],
                                                  "kind": "Field",
                                                  "name": Object {
                                                    "kind": "Name",
                                                    "value": "name",
                                                  },
                                                  "selectionSet": undefined,
                                                },
                                                Object {
                                                  "alias": undefined,
                                                  "arguments": Array [],
                                                  "directives": Array [],
                                                  "kind": "Field",
                                                  "name": Object {
                                                    "kind": "Name",
                                                    "value": "apiGroup",
                                                  },
                                                  "selectionSet": undefined,
                                                },
                                                Object {
                                                  "alias": undefined,
                                                  "arguments": Array [],
                                                  "directives": Array [],
                                                  "kind": "Field",
                                                  "name": Object {
                                                    "kind": "Name",
                                                    "value": "kind",
                                                  },
                                                  "selectionSet": undefined,
                                                },
                                              ],
                                            },
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "raw",
                                            },
                                            "selectionSet": undefined,
                                          },
                                        ],
                                      },
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "clusterCompliant",
                                      },
                                      "selectionSet": undefined,
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                          "variableDefinitions": Array [
                            Object {
                              "defaultValue": undefined,
                              "directives": Array [],
                              "kind": "VariableDefinition",
                              "type": Object {
                                "kind": "NonNullType",
                                "type": Object {
                                  "kind": "NamedType",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "String",
                                  },
                                },
                              },
                              "variable": Object {
                                "kind": "Variable",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "name",
                                },
                              },
                            },
                            Object {
                              "defaultValue": undefined,
                              "directives": Array [],
                              "kind": "VariableDefinition",
                              "type": Object {
                                "kind": "NamedType",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "String",
                                },
                              },
                              "variable": Object {
                                "kind": "Variable",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "namespace",
                                },
                              },
                            },
                          ],
                        },
                      ],
                      "kind": "Document",
                      "loc": Object {
                        "end": 800,
                        "start": 0,
                      },
                    }
                  }
                  variables={
                    Object {
                      "name": "test-policy",
                      "namespace": "test-namespace",
                    }
                  }
                >
                  <Spinner
                    className="patternfly-spinner"
                  >
                    <span
                      aria-valuetext="Loading..."
                      className="pf-c-spinner pf-m-xl patternfly-spinner"
                      role="progressbar"
                    >
                      <span
                        className="pf-c-spinner__clipper"
                      />
                      <span
                        className="pf-c-spinner__lead-ball"
                      />
                      <span
                        className="pf-c-spinner__tail-ball"
                      />
                    </span>
                  </Spinner>
                </Query>
              </PolicyDetailsTab>
            </Connect(PolicyDetailsTab)>
          </withRouter(Connect(PolicyDetailsTab))>
        </div>
      </Page>
    </PolicyDetailSubRouter>
  </Connect(PolicyDetailSubRouter)>
</withRouter(Connect(PolicyDetailSubRouter))>
`;
